### PR TITLE
test(symcollector): Add tests for filter options

### PR DESCRIPTION
**Pull request by iker-barriocanal**
- Clean up tests for the base artifact provider.
- Add tests to cover more edge-cases, such as having strings, regex, and undefined on the same call.
- Other minor changes.

### CODE REVIEW

File: src/artifact_providers/__tests__/base.test.ts [hunk 1]
```diff
@@ -9,33 +9,73 @@ describe('parseFilterOptions', () => {
...
-    expect(parsedFilters).not.toHaveProperty('includeNames');
-    expect(parsedFilters).not.toHaveProperty('excludeNames');
+
+    if (includeNames) {
+      expect(parsedFilters.includeNames).toStrictEqual(
+        stringToRegexp(includeNames)
+      );
+    } else {
+      expect(parsedFilters.includeNames).not.toBeDefined();
+    }
```
Comment (BYK) [inline src/artifact_providers/__tests__/base.test.ts:30]:
```suggestion
    expect(parsedFilters.includeNames).toStrictEqual(
      includeNames && stringToRegexp(includeNames)
    );
```

Comment (BYK) [reply src/artifact_providers/__tests__/base.test.ts:30]: Or

```suggestion
    expect(parsedFilters.includeNames).toStrictEqual(
      includeNames ? stringToRegexp(includeNames) : undefined
    );
```

File: src/artifact_providers/__tests__/base.test.ts [hunk 2]
```diff
@@ -9,33 +9,73 @@ describe('parseFilterOptions', () => {
...
   });
 
-  test('string properties', () => {
-    const stringFilter = '/testFilter/';
+  test.each([
+    [undefined, undefined],
+    [undefined, new RegExp('onlyExclude')],
+    [new RegExp('onlyInclude'), undefined],
+    [new RegExp('include'), new RegExp('exclude')],
+  ])('undefined and regex properties', (includeNames, excludeNames) => {
```
Comment (BYK) [inline src/artifact_providers/__tests__/base.test.ts:45]: I think you can fold these cases into the one above (the `undefined` ones are already redundant) and add a typecheck to the expectation:

```ts
    expect(parsedFilters.includeNames).toBeInstanceOf(RegExp);
```

File: src/artifact_providers/__tests__/base.test.ts [hunk 3]
```diff
@@ -9,33 +9,73 @@ describe('parseFilterOptions', () => {
...
-      includeNames: regexFilter,
+  test('string and regex properties', () => {
+    const strIncNames: RawFilterOptions = {
+      includeNames: '/string/',
+      excludeNames: new RegExp('regexp'),
     };
-    const parsedFilters = parseFilterOptions(rawFilters);
-    expect(parsedFilters.includeNames).toStrictEqual(regexFilter);
+    const parsedStrInc = parseFilterOptions(strIncNames);
+    expect(parsedStrInc.includeNames).toStrictEqual(stringToRegexp('/string/'));
```
Comment (BYK) [inline src/artifact_providers/__tests__/base.test.ts:70]: Using `stringToRegexp` for tests is a bit implementation-specific. A better way would be to actually check the filters to match a certain string. It would also make your test case above a lot clearer.

File: src/targets/symbolCollector.ts [hunk 4]
```diff
@@ -36,13 +36,12 @@ export class SymbolCollector extends BaseTarget {
     artifactProvider: BaseArtifactProvider
   ) {
     super(config, artifactProvider);
+    // The Symbol Collector should be available in the path
+    checkExecutableIsPresent(SYM_COLLECTOR_BIN_NAME);
```
Comment (BYK) [inline src/targets/symbolCollector.ts:40]: How is this relevant to this PR?

**Review by BYK — CHANGES_REQUESTED** [review]

File: src/artifact_providers/__tests__/base.test.ts (hunk 2 above)
Comment (iker-barriocanal) [reply src/artifact_providers/__tests__/base.test.ts:45]: Merging them would mean adding another type check before each expectation since having a string or regexp in the input is not the same. Separating them into two different test cases makes it easier to read and also to identify the failing test case (if any).

File: src/artifact_providers/__tests__/base.test.ts [hunk 5]
```diff
@@ -9,33 +8,44 @@ describe('parseFilterOptions', () => {
...
+  test.each([
+    [undefined, undefined],
+    [undefined, '/exclude/'],
+    [undefined, new RegExp('exclude')],
+    ['/include/', undefined],
+    [new RegExp('include'), undefined],
+    ['/include/', '/exclude/'],
+    ['/include/', new RegExp('exclude')],
+    [new RegExp('include'), '/exclude/'],
+    [new RegExp('include'), new RegExp('exclude')],
```
Comment (BYK) [inline src/artifact_providers/__tests__/base.test.ts:20]: why not the short form of regular expressions?

```suggestion
    [undefined, /exclude/],
    ['/include/', undefined],
    [/include/, undefined],
    ['/include/', '/exclude/'],
    ['/include/', /exclude/],
    [/include/, '/exclude/'],
    [/include/, /exclude/],
```

File: src/artifact_providers/__tests__/base.test.ts [hunk 6]
```diff
@@ -9,33 +8,44 @@ describe('parseFilterOptions', () => {
...
-  });
+      if (includeNames) {
+        expect(parsedFilters.includeNames).toStrictEqual(
+          typeof includeNames === 'string'
+            ? new RegExp(includeNames.slice(1, -1))
+            : includeNames
+        );
+      } else {
+        expect(parsedFilters.includeNames).not.toBeDefined();
+      }
```
Comment (BYK) [inline src/artifact_providers/__tests__/base.test.ts:38]:
```suggestion
    expect(parsedFilters.includeNames).toStrictEqual(includeNames && /include/);
```

File: src/artifact_providers/__tests__/base.test.ts [hunk 7]
```diff
@@ -9,33 +8,44 @@ describe('parseFilterOptions', () => {
...
+      if (excludeNames) {
+        expect(parsedFilters.excludeNames).toStrictEqual(
+          typeof excludeNames === 'string'
+            ? new RegExp(excludeNames.slice(1, -1))
+            : excludeNames
+        );
+      } else {
+        expect(parsedFilters.excludeNames).not.toBeDefined();
+      }
+    }
```
Comment (BYK) [inline src/artifact_providers/__tests__/base.test.ts:49]:
```suggestion
    expect(parsedFilters.excludeNames).toStrictEqual(/exclude/);
```

**Review by BYK — APPROVED** [review]
🚀
